pkg/commands/plan/plan_init.go

```diff
@@ -0,0 +1,249 @@
+	})
+
+	f.BoolVar(&cli.BoolVar{
+		Name:    "json",
+		Target:  &c.flagJSON,
+		Default: false,
+		Usage:   "Print the output directories in JSON format.",
+	})
+
+	return set
+}
+
+func (c *PlanInitCommand) Run(ctx context.Context, args []string) error {
+	logger := logging.FromContext(ctx)
+
+	f := c.Flags()
+	if err := f.Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+
+	args = f.Args()
+	if len(args) > 0 {
+		return fmt.Errorf("unexpected arguments: %q", args)
+	}
+
+	c.actions = githubactions.New(githubactions.WithWriter(c.Stdout()))
+	actionsCtx, err := c.actions.Context()
+	if err != nil {
+		return fmt.Errorf("failed to load github context: %w", err)
+	}
+
+	c.cfg = &Config{}
+	if err := c.cfg.MapGitHubContext(actionsCtx); err != nil {
+		return fmt.Errorf("failed to load github context: %w", err)
+	}
+	logger.Debugw("loaded configuration", "config", c.cfg)
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed to get current working directory: %w", err)
+	}
+	c.workingDir = cwd
+
+	c.gitClient = git.NewGitClient(cwd)
+	c.githubClient = github.NewClient(
+		ctx,
+		c.GitHubFlags.FlagGitHubToken,
+		github.WithInitialRetryDelay(c.RetryFlags.FlagInitialRetryDelay),
+		github.WithMaxRetries(c.RetryFlags.FlagMaxRetries),
+		github.WithMaxRetryDelay(c.RetryFlags.FlagMaxRetryDelay),
+	)
+
+	if len(c.flagDirectories) == 0 {
+		c.flagDirectories = append(c.flagDirectories, cwd)
+	}
+
+	logger.Debugln("finding entrypoint directories..")
+
+	dirs, err := terraform.GetEntrypointDirectories(c.flagDirectories)
+	if err != nil {
+		return fmt.Errorf("failed to find terraform directories: %w", err)
+	}
+
+	c.entrypoints = dirs
+	logger.Debugf("terraform entrypoint directories: %s", c.entrypoints)
+
+	return c.Process(ctx)
+}
+
+// Process handles the main logic for the Guardian plan init process.
+func (c *PlanInitCommand) Process(ctx context.Context) error {
+	logger := logging.FromContext(ctx)
+	logger.Debugln("starting Guardian plan init")
+
+	if c.GitHubFlags.FlagGitHubAction && c.flagDeleteOutdatedComments {
+		logger.Debugln("removing outdated comments...")
+		if err := c.deleteOutdatedComments(ctx, c.GitHubFlags.FlagGitHubOwner, c.GitHubFlags.FlagGitHubRepo, c.flagPullRequestNumber); err != nil {
+			return fmt.Errorf("failed to delete outdated comments: %w", err)
+		}
+	}
+
+	diffDirs, err := c.gitClient.DiffDirsAbs(ctx, c.flagDestRef, c.flagSourceRef)
+	if err != nil {
+		return fmt.Errorf("failed to find git diff directories: %w", err)
+	}
+	logger.Debugf("changed directories: %+v", diffDirs)
+
+	targetDirs := util.GetSliceIntersection(c.entrypoints, diffDirs)
+	logger.Debugf("target directories: %+v", targetDirs)
+
+	if c.flagJSON {
+		outJSON, err := json.Marshal(targetDirs)
+		if err != nil {
+			return fmt.Errorf("failed to create json string: %w", err)
+		}
+		c.Outf(string(outJSON))
+	} else {
+		for _, dir := range targetDirs {
+			c.Outf("%s", dir)
```

Comment:
Do you need a newline after this?

Reply:
Outf handles that for you....which I am not sure I agree with but for now, it works.

https://github.com/abcxyz/pkg/blob/main/cli/command.go#L302